### PR TITLE
Fix no-cmac

### PR DESCRIPTION
**Pull request by mattcaswell**
Don't include files that we don't want to build

### CODE REVIEW

File: providers/common/macs/build.info [hunk 1]
```diff
@@ -1,4 +1,8 @@
-$COMMON=cmac_prov.c gmac_prov.c hmac_prov.c kmac_prov.c
+IF[{- !$disabled{cmac} -}]
+  $COMMON=cmac_prov.c gmac_prov.c hmac_prov.c kmac_prov.c
+ELSE
+  $COMMON=gmac_prov.c hmac_prov.c kmac_prov.c
+ENDIF
```
Comment (levitte) [inline providers/common/macs/build.info]: This example might give you a better idea on how not to repeat things so much :wink:

```
  $COMMON=$COMMON cmac_prov.c
```

Comment (mattcaswell) [reply providers/common/macs/build.info]: Fixed.

**Comment by mattcaswell** [comment]
@levitte - please take a look at the updated version.

**Comment by mspncp** [comment]
@mattcaswell it seems like every once in a while you have to fix some broken `no-xxx` options. Would it be feasible to test all (or just the most common) `no-xxx` options on a regular base? For example by having dedicated CI jobs which only run on a daily[, weekly, manual,...] schedule?

**Comment by mattcaswell** [comment]
> Would it be feasible to test all (or just the most common) no-xxx options on a regular base? For example by having dedicated CI jobs which only run on a daily[, weekly, manual,...] schedule?

We already do...that's why I know they need to be fixed! See the run-checker script in the tools directory. Failed build notifications get sent to openssl-commits.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**